ext/websocket/02_websocketstream.js

```diff
@@ -143,6 +147,7 @@
               ? ArrayPrototypeJoin(options.protocols, ", ")
               : "",
             cancelHandle: cancelRid,
+            headers: options.headers,
```

Comment:
You need to do `new Headers()` with the `options.headers` first. Then you should get the header list from the `Headers` and pass that to rust. Currently this code will break when a user passes an object instead of a `[string, string][]`.